src/rgw/rgw_sal_motr.cc

```diff
@@ -392,6 +393,10 @@ int MotrUser::store_user(const DoutPrefixProvider* dpp,
   obj_version& obj_ver = objv_tr.read_version;
 
   ldpp_dout(dpp, 20) << "Store_user(): User = " << info.user_id.id << dendl;
+  if (!info.user_email.empty()) {
+     MotrEmailInfo MGWEmailInfo(info.user_id.id, info.user_email);
+     store->store_email_info(dpp, y, MGWEmailInfo);
+  }
```

Comment:
We should add email entry in email index only if user info is successfully stored in user index. If user info fails to store due to some issue, we will have a stale entry in email index table. Please move this block after user store is complete, i.e. after line 426 / after below code:
```
 if (rc == 0) {
    objv_tracker.read_version = obj_ver;
    objv_tracker.write_version = obj_ver;
  }
```